src/http/request.rs

```diff
@@ -132,6 +135,41 @@ impl<B> fmt::Debug for Request<B> {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<Request> for http_types::Request<Body> {
```

Comment:
Unfortunately not, since the only reason Rust lets me impl this is due to the fact that part of the impl receiver is from this crate (since both the trait as well as the receiver of the impl are from other crates).

Reply:
Ah dang, I see what you mean, it can't be done because of the generic + `From` + target is not a crate local.

- It can stay as it is, using the default generic of `hyper::Request`, and just not have a conversion when one has `hyper::Request<SomethingElse>`. In this case, `hyper::Request<B>` can grow an `into_compat` method to compensate...
- We can have an `Into` instead that does allow the generic, but then `http::Request::from(hyper)` doesn't work...

I'm not certain which is preferable, actually. Probably depends on which case is more common (using `From::from`, or `hyper::Request<NotDefaultBody>`).

Reply:
To get an idea where you stand on this, do you see the panic-ing of the conversion functions as a concern at all, or do you think it's a non-issue? Introducing `from` methods that panic has been bothering me more and more when thinking about it during the last couple of days, and at least personally I would feel a lot more confident about the quality of my contribution if the conversion methods could safely signal an error w/o resorting to `panic!`.

Reply:
I'm personally fine with them panicking for now. It is an opt-in compat layer, and any panics should be horribly wrong, unexpected bugs. If `TryFrom` were stable, I'd be fine with using that, but it's not.

If you'd rather, `try_into_http` inherent methods can be added, to allow non-panicking methods.

Reply:
In that case, I'd be fine with leaving it like it is. Or did you want me to add the generic `into_compat` method before we merge? Anything else you'd like to see changed?